src/org/usfirst/frc/team930/robot/subsystems/SwerveDrive.java

```diff
@@ -0,0 +1,431 @@
+package org.usfirst.frc.team930.robot.subsystems;
+
+//import edu.wpi.first.wpilibj.SpeedController;
+//import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.command.Subsystem;
+
+public class SwerveDrive extends Subsystem {
+
+	/*
+	 * Team 930 SwerveDrive
+	 * 
+	 * Heading of 0 degrees is straight forward to the robot. Interface for
+	 * other objects are the INTERFACE METHODS, and private calculation methods
+	 * are the CALC METHODS.
+	 * 
+	 * For the field centric utilization of the code, you must calibrate the
+	 * robot so that it faces perpendicular to the baseline of the field.
+	 * Heading zero is that way
+	 */
+
+	// DECLARATIONS OF VARIABLES AND OTHER THINGS THE CODE MIGHT FIND USEFUL
+
+	// Time Components
+	final long UPDATE_TIME = 5; // time between updates to the robot (ms)
+	long updateStartTime; // time of this update's start
+	long updateEndTime; // time of the last update's end
+	long timeSinceLastUpdate; // time since the last update, in seconds
+
+	// SpeedController Components - if possible to implement
+	// SpeedController TR, TL, BR, BL;
+
+	// Field Centric Specific Components
+	private boolean isFieldcentric; // are we doin' field centric calculations?
+
+	private double heading, lastHeading; // field centric headings
+
+	// Joysticks
+	Joystick translationStick, headingStick;
+
+	// Robot Specs
+	private double width, length; // length and width of the robot
+
+	// Calculation Components
+	final double ANGLE_CONVERSION = 180 / Math.PI;
+
+	// CONSTRUCTORS - eventually get them all to accept passed SpeedControllers
+
+	public SwerveDrive(double length, double width) {
+		this.width = width;
```

Comment:
Instead of duplicating code between these two constructors, call

``` this(length, width, false) ```

in here.